### PR TITLE
Create CODE_OF_CONDUCT.md

### PR DESCRIPTION
**Pull request by eliabsisay**
Add code of conduct to android open source repo

### CODE REVIEW

**Review by fractalwrench — APPROVED** [review]

File: CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
+# Contributor Covenant Code of Conduct
+
+## Our Pledge
+
+We as members, contributors, and leaders pledge to make participation in our
+community a harassment-free experience for everyone, regardless of age, body
+size, visible or invisible disability, ethnicity, sex characteristics, gender
```
Comment (bidetofevil) [inline CODE_OF_CONDUCT.md:7]:
```suggestion
size, visible or invisible disability, ethnicity, sex, gender
```

Comment (bidetofevil) [reply CODE_OF_CONDUCT.md:7]: sex characteristics sound weird. Simply sex is enough I think.

File: CODE_OF_CONDUCT.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+## Enforcement Responsibilities
+
+Community leaders are responsible for clarifying and enforcing our standards of
+acceptable behavior and will take appropriate and fair corrective action in
+response to any behavior that they deem inappropriate, threatening, offensive,
+or harmful.
+
+Community leaders have the right and responsibility to remove, edit, or reject
+comments, commits, code, wiki edits, issues, and other contributions that are
```
Comment (bidetofevil) [inline CODE_OF_CONDUCT.md:47]:
```suggestion
comments, commits, code, issues, and other contributions that are
```

Comment (bidetofevil) [reply CODE_OF_CONDUCT.md:47]: We don't have a wiki so I think putting this in there make it seem like it's copied from somewhere.

File: CODE_OF_CONDUCT.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+All community leaders are obligated to respect the privacy and security of the
+reporter of any incident.
+
+## Enforcement Guidelines
+
+Community leaders will follow these Community Impact Guidelines in determining
+the consequences for any action they deem in violation of this Code of Conduct:
+
+### 1. Correction
```
Comment (bidetofevil) [inline CODE_OF_CONDUCT.md:73]: Is this list supposed to be an escalation? Or just a mapping of severity of violation to consequences?

Comment (bidetofevil) [inline CODE_OF_CONDUCT.md:73]: Not sure about calling this step `Correction` - unless it's accepted terminology in this space. It feels more like an unofficial warning  or "a talking to"

File: CODE_OF_CONDUCT.md [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+**Community Impact**: Use of inappropriate language or other behavior deemed
+unprofessional or unwelcome in the community.
+
+**Consequence**: A private, written warning from community leaders, providing
+clarity around the nature of the violation and an explanation of why the
+behavior was inappropriate. A public apology may be requested.
+
+### 2. Warning
+
+**Community Impact**: A violation through a single incident or series
```
Comment (bidetofevil) [inline CODE_OF_CONDUCT.md:84]: This doesn't sufficiently differentiate itself from 1.

File: CODE_OF_CONDUCT.md [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+**Consequence**: A warning with consequences for continued behavior. No
+interaction with the people involved, including unsolicited interaction with
+those enforcing the Code of Conduct, for a specified period of time. This
+includes avoiding interactions in community spaces as well as external channels
+like social media. Violating these terms may lead to a temporary or
+permanent ban.
+
+### 3. Temporary Ban
+
+**Community Impact**: A serious violation of community standards, including
```
Comment (bidetofevil) [inline CODE_OF_CONDUCT.md:96]: 2 mentions "a series" too - how is this different than that?

File: CODE_OF_CONDUCT.md [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+
+**Consequence**: A temporary ban from any sort of interaction or public
+communication with the community for a specified period of time. No public or
+private interaction with the people involved, including unsolicited interaction
+with those enforcing the Code of Conduct, is allowed during this period.
+Violating these terms may lead to a permanent ban.
+
+### 4. Permanent Ban
+
+**Community Impact**: Demonstrating a pattern of violation of community
```
Comment (bidetofevil) [inline CODE_OF_CONDUCT.md:107]: Should we require an escalating consequence, e.g. first get warning, then get temporary ban, then perma ban, just so we can demonstrate a refusal to change behaviour?

**Review by bidetofevil — COMMENTED** [review]
Looks good overall. Minor language tweaks, and some comments about a more clear differentiation of the levels and a statement whether these are escalations or just a map of seriousness to consequence of behaviour.

File: CODE_OF_CONDUCT.md (hunk 3 above)
Comment (eliabsisay) [reply CODE_OF_CONDUCT.md:73]: I think a bit of both, if something is clearly obscene and intentional we can jump directly to banning the user. We have a lot of latitude.

File: CODE_OF_CONDUCT.md (hunk 4 above)
Comment (eliabsisay) [reply CODE_OF_CONDUCT.md:84]: I agree - going to just make Warning the first step

File: CODE_OF_CONDUCT.md (hunk 5 above)
Comment (eliabsisay) [reply CODE_OF_CONDUCT.md:96]: I think thats intended to give us leeway depending on the severity of the infraction

File: CODE_OF_CONDUCT.md (hunk 6 above)
Comment (eliabsisay) [reply CODE_OF_CONDUCT.md:107]: I dont think thats necessary - if something is so flagrant that it deserves and immediate permanent ban i want to maintain optionality in doint that

**Pull request merged**